service/controller/v17/encrypter/spec.go

```diff
@@ -44,7 +29,16 @@ type Encrypter interface {
 	IsKeyNotFound(error) bool
 }
 
+type Resource interface {
+	EnsureCreatedEncryptionKey(context.Context, v1alpha1.AWSConfig) error
+	EnsureDeletedEncryptionKey(context.Context, v1alpha1.AWSConfig) error
+}
+
 type RoleManager interface {
 	EnsureCreatedAuthorizedIAMRoles(context.Context, v1alpha1.AWSConfig) error
 	EnsureDeletedAuthorizedIAMRoles(context.Context, v1alpha1.AWSConfig) error
 }
+
+type TLSManager interface {
+	EncryptTLSAssets(ctx context.Context, customObject v1alpha1.AWSConfig, assets legacy.AssetsBundle) (*legacy.CompactTLSAssets, error)
+}
```

Comment:
Again this is more encapsulated now, goes toward direction of making it a resource.